pkg/cmd/pr/merge/merge.go

```diff
@@ -346,17 +376,6 @@ func (m *mergeContext) deleteLocalBranch() error {
 
 	// branch the command was run on is the same as the pull request branch
 	if currentBranch == m.pr.HeadRefName {
-		// if the target branch of the PR is not known, set the current branch to the
-		// default branch of the repository
-		targetBranch := m.pr.BaseRefName
```

Comment:
with the addition of `baseRefName` to `opts.Finder.Find` below, `m.pr.BaseRefName` can no longer be empty.

From the [graphql documentation](https://docs.github.com/en/graphql/reference/objects#pullrequest):
```
baseRefName (String!)
Identifies the name of the base Ref associated with the pull request, even if the ref has been deleted.
```

if the baseRefName is instead fetched on the PR as 
```
baseRef {
    name
}
```
it can be `null` if the base ref has been deleted